llvm/lib/Target/AArch64/AArch64ISelLowering.cpp

```diff
@@ -5705,11 +5705,11 @@ SDValue AArch64TargetLowering::LowerSTORE(SDValue Op,
     // legalization will break up 256 bit inputs.
     ElementCount EC = MemVT.getVectorElementCount();
     if (StoreNode->isNonTemporal() && MemVT.getSizeInBits() == 256u &&
-        EC.isKnownEven() &&
-        ((MemVT.getScalarSizeInBits() == 8u ||
-          MemVT.getScalarSizeInBits() == 16u ||
-          MemVT.getScalarSizeInBits() == 32u ||
-          MemVT.getScalarSizeInBits() == 64u))) {
+        EC.isKnownEven() && DAG.getDataLayout().isLittleEndian() &&
```

Comment:
Can you please explain why this logic here is not constrained to little endian?

From the PR description, I assumed that non-temporal stores would be now enabled for both endiannesses.

Reply:
Non-temporal stores are a performance optimization, we can always drop the "non-temporal" part and just store them normally. The LDNP nodes are already handled this same way, this brings the STNP in-line.
If you see the tests there is a lot less shuffling which should be better for performance.